bgpd/bgp_updgrp_adv.c

```diff
@@ -533,6 +533,14 @@ void bgp_adj_out_unset_subgroup(struct bgp_node *rn,
 		if (adj->adv)
 			bgp_advertise_clean_subgroup(subgrp, adj);
 
+		/* If default originate is enabled and the route is default
+		 * route, do not send withdraw
+		 */
+		if (CHECK_FLAG(subgrp->sflags,
+			       SUBGRP_STATUS_DEFAULT_ORIGINATE) &&
+		    is_default_prefix(&rn->p))
+			return;
```

Comment:
Also if we have default-originate before and then default-originate with route-map option is given to match on some prefix which is not present then already advertised default route needs to be withdrawn and that will not happen due to above check.

Reply:
default route advertisement using default originate is handled seperately in subgroup_default_update_packet() and deletion in subgroup_default_withdraw_packet()

When default route is advertised previously using redistribute command and then default originate is added, it is sufficient to clear the adj out entry since the newly advertised copy of default route will serve as implicit withdraw to the peer, the receiver will replace old copy with the new copy